controlplane/eks/api/v1alpha4/types.go

```diff
@@ -214,3 +214,59 @@ const (
 	// control plane and managed node groups.
 	SecurityGroupCluster = infrav1.SecurityGroupRole("cluster")
 )
+
+type OIDCIdentityProviderConfig struct {
+
+	// This is also known as audience. The ID for the client application that makes
+	// authentication requests to the OpenID identity provider.
+	// +kubebuilder:validation:Required
+	ClientID *string `json:"clientId,omitempty"`
```

Comment:
For the required fields you could probably get away with no using a pointer.

Reply:
Ack